src/server/project.js

```diff
@@ -317,6 +326,11 @@ Project.prototype._getProjectState = function() {
             var pluginsJsonPath = path.join(projectRoot, 'plugins', platform + '.json');
             return Q.nfcall(fs.readFile, pluginsJsonPath);
         })
+        .fail(function () {
+            // an error ocurred trying to read the file for the current platform,
+            // set the pluginsList to indicate a "bad state".
+            newState.pluginList = Project._PLUGINS_LIST_BAD_STATE;
```

Comment:
I think that this isn't quite right; since you don't rethrow an error this will pass void to the next promise handler, which will then cause the `fileContent.toString()` to fail and go to the error handler, but I'm not sure what the point of directly assigning to `newState.pluginList` here is.

I'd prefer to either `return '{}';` to convert a missing file into an empty set of installed plugins, or to put this failure handler after the next then so we only assign to `newState.pluginList` once.